tutorials/Dashboards_and_Low-Code_Apps/GQI/Custom_Operator_Tutorial.md

```diff
@@ -0,0 +1,168 @@
+
+[GQIMetaData(Name = "Balance & obfuscate")]
+public class MyOperator : IGQIColumnOperator, IGQIRowOperator, IGQIInputArguments
+{
+    // 3 column input arguments: income, expenses, and name 
+    private readonly GQIColumnDropdownArgument _incomeColumnArgument;
+    private readonly GQIColumnDropdownArgument _expensesColumnArgument;
+    private readonly GQIColumnDropdownArgument _nameColumnArgument;
+
+    // 3 variables to store the given argument values
+    private GQIColumn _incomeColumn;
+    private GQIColumn _expensesColumn;
+    private GQIColumn _nameColumn;
+
+    // new balance output column
+    private readonly GQIIntColumn _balanceColumn;
+
+    public MyOperator()
+    {
+        // initialize input column arguments
+        _incomeColumnArgument = new GQIColumnDropdownArgument("Income")
+        {
+            IsRequired = true,
+            Types = new GQIColumnType[] { GQIColumnType.Int }
+        };
+        _expensesColumnArgument = new GQIColumnDropdownArgument("Expenses")
+        {
+            IsRequired = true,
+            Types = new GQIColumnType[] { GQIColumnType.Int }
+        };
+        _nameColumnArgument = new GQIColumnDropdownArgument("Name")
+        {
+            IsRequired = true,
+            Types = new GQIColumnType[] { GQIColumnType.String }
+        };
+
+        // new balance output column
+        _balanceColumn = new GQIIntColumn("Balance");
+    }
+
+    public GQIArgument[] GetInputArguments()
+    {
+        // define input arguments
+        return new[] {
+            _incomeColumnArgument,
+            _expensesColumnArgument,
+            _nameColumnArgument
+        };
+    }
+
+    public OnArgumentsProcessedOutputArgs OnArgumentsProcessed(OnArgumentsProcessedInputArgs args)
+    {
+        // store given argument values
+        _incomeColumn = args.GetArgumentValue(_incomeColumnArgument);
+        _expensesColumn = args.GetArgumentValue(_expensesColumnArgument);
+        _nameColumn = args.GetArgumentValue(_nameColumnArgument);
+
+        return default;
+    }
+
+    public void HandleColumns(GQIEditableHeader header)
+    {
+        // new balance output column
+        header.AddColumns(_balanceColumn);
+    }
+
+    public void HandleRow(GQIEditableRow row)
+    {
+        // calculate and set balance
+        var income = row.GetValue<int>(_incomeColumn);
+        var expenses = row.GetValue<int>(_expensesColumn);
+        var balance = income - expenses;
+        row.SetValue(_balanceColumn, balance);
+
+        // obfuscate name
+        var name = row.GetValue<string>(_nameColumn);
+        var obfuscatedName = ObfuscateName(name);
+        row.SetValue(_nameColumn, obfuscatedName);
+    }
+
+    private string ObfuscateName(string name)
+    {
+        var obfuscatedName = "";
+        // apply an obfuscation transformation
+        foreach (var part in name.Split())
+            obfuscatedName += $"{part[0]}.";
+        return obfuscatedName;
+    }
+}
+```
```

Comment:
How do you all of a sudden arrive at this implementation? What do users need to do to achieve this? Is this the end result already and does that mean the rest of the tutorial (available in the internal documentation) only offers use cases (i.e. examples of what do with this custom operator)? 

You start off use case 1 with "Let's try out our custom operator and create a query.". However, we never explained how users can create the custom operator in the first place.